### DISCUSSION TITLE
autoDisplayFirst

### DISCUSSION THREAD

**Discussion by adulfan** [primefaces/community#1383] [PrimeNG] [answered]
Since autoDisplayFirst was deprecated what are the options/solutions to display first option on the dropdown and autocomplete components? adding autoDisplayFirst worked up to 17.1 version. Thank you

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

`autoDisplayFirst` causes some maintenance issues in the component logic. You can simply set an initial value for the component.

```

// html
<p-dropdown [options]="cities" [(ngModel)]="selectedCity" optionLabel="name" [showClear]="true" placeholder="Select a City"></p-dropdown>

// ts 

selectedCity = this.cities[0] // etc. whatever you want
```

**Reply by adulfan** [reply]
This is with ngModel, I am on Angular 17 and using Reactive forms. What would be an option for the components and I am also wrapping it in custom to be able to pass some properties, like label, would be nice if that came OTB on the plugin, where you can pass a label text into all form elements.
